client/state/sites/domains/assembler.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import moment from 'moment';
+
+const setMomentType = date => {
+	return date
+		? moment( date ).startOf( 'day' )
+		: date;
+};
+
+export const createSiteDomainObject = domain => {
```

Comment:
I see a few other assembler.js object mapping usages in the state tree. If we like this pattern we should probably add some reasoning in https://github.com/Automattic/wp-calypso/blob/master/docs/our-approach-to-data.md 

cc @drewblaisdell @aduth